mrbgems/mruby-regexp/test/regexp.rb

```
@@ -1282,6 +1291,59 @@ class StringMatchHelperOverride < String
+  # which is what the pre-scan buys: the parser reaches it before it has seen
+  # the declaration that decides the question
+  md = /(a)(?<b>b)/.match("ab")
+  assert_equal 2, md.size
+  assert_equal ["ab", "b"], md.to_a
+  assert_equal ["b"], md.captures
+  assert_equal "b", md[1]
+  assert_equal "b", md[:b]
+
+  # the shrunken count is what $2, $+ and a \2 in a replacement read
+  "ab" =~ /(?<a>a)(b)/
+  assert_nil $2
```

RuboCop (1.88.2)

[warning] $2 is out of range (1 regexp capture group detected).

(Lint/OutOfRangeRegexpRef)